generic/piranha-tree-sitter/src/cleanup_rules/java/rules.toml

```diff
@@ -0,0 +1,540 @@
+        right: [(false) (parenthesized_expression (false))]
+    )
+@binary_expression)"""
+replace_node = "binary_expression"
+replace = "@lhs"
+groups = ["boolean_expression_simplify"]
+
+# Before: 
+#  false || abc()
+# After :
+#  abc()
+#
+[[rules]]
+name = "simplify_false_or_something"
+query = """
+(
+    (binary_expression
+        left : [(false) (parenthesized_expression (false))]
+        operator:"||"
+        right: (_)* @rhs
+    )
+@binary_expression)"""
+replace_node = "binary_expression"
+replace = "@rhs"
+groups = ["boolean_expression_simplify"]
+
+# Before :
+#  {
+#    something();
+#    return 10;
+#    somethingMore();  
+#    return 100;
+#    
+#  } 
+# After :
+#  {
+#    something();
+#    return 10;
+#  } 
+#
+[[rules]]
+name = "delete_all_statements_after_return"
+query = """(
+        (block  ((statement)* @pre)
+         ((return_statement) @r)
+         ((statement)+ @post)) @b)"""
+replace_node = "post"
+replace = ""
+
+# Before :
+#  true ? abc() : def();
+# After :
+#  abc()
+#
+[[rules]]
+name = "simplify_ternary_operator_true"
+query = """
+(
+    (ternary_expression condition: [(true) (parenthesized_expression (true))]
+        consequence: (_)* @consequence
+        alternative: (_)* @alternative)
+@ternary_expression)"""
+replace_node = "ternary_expression"
+replace = "@consequence"
+groups = ["if_cleanup"]
+
+# Before :
+#  false ? abc() : def();
+# After :
+#  def()
+#
+[[rules]]
+name = "simplify_ternary_operator_false"
+query = """
+(
+    (ternary_expression condition: [(false) (parenthesized_expression (false))]
+        consequence: (_)* @consequence
+        alternative: (_)* @alternative)
+@ternary_expression)"""
+replace_node = "ternary_expression"
+replace = "@alternative"
+groups = ["if_cleanup"]
+
+# This rule is part (and entry point) for the inline local variable cleanup. Example: 
+# 
+# Before :
+# public foo someMethod() { 
+#  boolean isFlag = true; 
+#  somethingElse();
+# }
+#  
+# After :
+# public foo someMethod() { 
+#  somethingElse();
+# }
+#
+# Before inlining a variable we need to ensure that the variable is not re assigned anywhere within the method body 
+# to a value which is not the same as the initlaizer of variable we are inlining..
+# Note the below negative example of this rule where the variable `isFlag` is re-assigned to the value `getFlagStatus()` while it's 
+# initlaizer is `true`. In such a scenario the variable `isFlag` should not be inlined.
+# public foo someMethod() { 
+#  boolean isFlag = true; 
+#  isFlag = getFlagStatus();
+#  somethingElse();
+# }
+#
+
+[[rules]]
+name = "delete_variable_declaration"
+query = """
+(
+((local_variable_declaration 
+				declarator: (variable_declarator 
+                					name: (_) @variable_name
+                					value: [(true) (false)] @init)) @variable_declaration)
+)
+"""
+replace_node = "variable_declaration"
+replace = ""
+groups = []
+# Check if there is no assignment where the variable @variable_name is 
```

Comment:
Thanks for the detailed comment. This greatly helps to understand the purpose of this constraint here.